### PR TITLE
slurm: 21.08.7.1 -> 21.08.8.2

### PR DESCRIPTION
**Pull request by posch**
https://www.schedmd.com/news.php

Fixes: CVE-2022-29500, CVE-2022-29501, CVE-2022-29502

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ajs124 — APPROVED** [review]

File: pkgs/servers/computing/slurm/default.nix [hunk 1]
```diff
@@ -49,15 +52,17 @@ stdenv.mkDerivation rec {
...
   ] ++ lib.optionals enableX11 [ xorg.xauth ];
 
   configureFlags = with lib;
     [ "--with-freeipmi=${freeipmi}"
       "--with-hwloc=${hwloc.dev}"
+      "--with-json=${json_c.dev}"
+      "--with-jwt=${libjwt}"
       "--with-lz4=${lz4.dev}"
       "--with-munge=${munge}"
-      "--with-zlib=${zlib}"
```
Comment (markuskowa) [inline pkgs/servers/computing/slurm/default.nix:60]: Is it still built with zlib properly?

Comment (posch) [reply pkgs/servers/computing/slurm/default.nix:60]: I don't think slurm is still using zlib.
There is no "--with-zlib" configure option.
grep doesn't show any potential uses of zlib.
The NEWS file indicates that zlib was replaced with lz4 in 16.05.0rc1.

Comment (markuskowa) [reply pkgs/servers/computing/slurm/default.nix:60]: ah, okay. Please remove it also from the inputs.

**Comment by markuskowa** [comment]
@GrahamcOfBorg eval

**Review by markuskowa — APPROVED** [review]
Thanks!

**Pull request merged**